src/gluonts/model/tft/_estimator.py

```diff
@@ -0,0 +1,294 @@
+
+from gluonts.core.component import DType, validated
+from gluonts.dataset.field_names import FieldName
+from gluonts.model.estimator import GluonEstimator
+from gluonts.model.predictor import RepresentableBlockPredictor
+from gluonts.support.util import copy_parameters
+from gluonts.time_feature import (
+    TimeFeature,
+    time_features_from_frequency_str,
+)
+from gluonts.transform import (
+    AddObservedValuesIndicator,
+    AddTimeFeatures,
+    AsNumpyArray,
+    Chain,
+    ExpectedNumInstanceSampler,
+    Transformation,
+    VstackFeatures,
+    SetField,
+)
+
+from ._transform import TFTInstanceSplitter
+from ._network import (
+    TemporalFusionTransformerTrainingNetwork,
+    TemporalFusionTransformerPredictionNetwork,
+)
+from ._engine import Trainer, QuantileForecastGenerator
+
+
+class TemporalFusionTransformerEstimator(GluonEstimator):
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        context_length: int,
+        prediction_length: Optional[int] = None,
+        trainer: Trainer = Trainer(),
+        hidden_dim: int = 32,
+        variable_dim: Optional[int] = None,
+        num_heads: int = 4,
+        num_outputs: int = 3,
+        num_instance_per_series: int = 100,
+        dropout_rate: float = 0.1,
+        time_features: Optional[List[TimeFeature]] = None,
+        static_cardinalities: Optional[Dict[str, int]] = None,
+        dynamic_cardinalities: Optional[Dict[str, int]] = None,
+        static_feature_dims: Optional[Dict[str, int]] = None,
+        dynamic_feature_dims: Optional[Dict[str, int]] = None,
+        past_dynamic_features: Optional[List[str]] = None,
+    ) -> None:
+        super(TemporalFusionTransformerEstimator, self).__init__(
+            trainer=trainer
+        )
+        assert (
+            prediction_length > 0
+        ), "The value of `prediction_length` should be > 0"
+        assert (
+            context_length is None or context_length > 0
+        ), "The value of `context_length` should be > 0"
+        assert dropout_rate >= 0, "The value of `dropout_rate` should be >= 0"
+
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.context_length = context_length or prediction_length
+        self.dropout_rate = dropout_rate
+        self.hidden_dim = hidden_dim
+        self.variable_dim = variable_dim or hidden_dim
+        self.num_heads = num_heads
+        self.num_outputs = num_outputs
+        self.num_instance_per_series = num_instance_per_series
+
+        self.time_features = time_features or time_features_from_frequency_str(
+            self.freq
+        )
+        self.static_cardinalities = static_cardinalities or {}
+        self.static_feature_dims = static_feature_dims or {}
+        self.dynamic_cardinalities = dynamic_cardinalities or {}
+        self.dynamic_feature_dims = dynamic_feature_dims or {}
+        self.past_dynamic_features = past_dynamic_features or []
```

Comment:
I think pydantic is able to handle this.

E.g.:

```python
time_features:List[TimeFeature] = [],
```

Normally you wouldn't do this in Python, but pydantic will return you a copy of the empty list, so this should work.

Reply:
fixed in abdc7114efd290505ea64ef943baebb656e7b5d0